processing/src/main/java/io/druid/query/lookup/LookupReferencesManager.java

```diff
@@ -308,26 +355,154 @@ private void takeSnapshot(Map<String, LookupExtractorFactoryContainer> lookupMap
     }
   }
 
-  private void loadSnapshotAndInitStateRef()
+  private void loadAllLookupsAndInitStateRef()
   {
-    if (lookupSnapshotTaker != null) {
-      ImmutableMap.Builder<String, LookupExtractorFactoryContainer> builder = ImmutableMap.builder();
-
-      final List<LookupBean> lookupBeanList = lookupSnapshotTaker.pullExistingSnapshot();
-      for (LookupBean lookupBean : lookupBeanList) {
-        LookupExtractorFactoryContainer container = lookupBean.getContainer();
+    if (!lookupConfig.getDisableLookupSync()) {
+      String tier = lookupListeningAnnouncerConfig.getLookupTier();
+      List<LookupBean> lookupBeanList = new ArrayList<>();
+      // Check if the coordinator is accessible
+      if (getCoordinatorUrl().isEmpty()) {
+        if (lookupSnapshotTaker != null) {
+          LOG.info("Coordinator is unavailable. Loading saved snapshot instead");
+          lookupBeanList = getLookupListFromSnapshot();
+        }
+      } else {
+        lookupBeanList = getLookupListFromCoordinator(tier);
+        if (lookupBeanList == null) {
+          LOG.info("Lookups not accessible from Coordinator. Loading saved snapshot instead");
+          lookupBeanList = getLookupListFromSnapshot();
+        }
+      }
+      if (!lookupBeanList.isEmpty()) {
+        ImmutableMap.Builder<String, LookupExtractorFactoryContainer> builder = ImmutableMap.builder();
+        ListeningScheduledExecutorService executorService = MoreExecutors.listeningDecorator(
+            Executors.newScheduledThreadPool(
+                lookupConfig.getNumLookupLoadingThreads(),
+                Execs.makeThreadFactory("LookupReferencesManager--%s")
+            )
+        );
+        try {
+          List<ListenableFuture<Map.Entry>> futures = new ArrayList<>();
+          LOG.info("Starting lookup loading process");
+          for (LookupBean lookupBean : lookupBeanList) {
+            futures.add(
+                executorService.submit(
+                    () -> {
+
+                      LookupExtractorFactoryContainer container = lookupBean.getContainer();
+                      if (container.getLookupExtractorFactory().start()) {
+                        LOG.info(
```

Comment:
can we add this log before calling start() ... "Starting lookup..." ? basically if some start() implementation is stuck for some reason then logs would tell which lookup is stuck on start().